server/tests/src/test/java/org/infinispan/server/resilience/ResilienceMaxRetryIT.java

```diff
@@ -34,39 +35,36 @@
 
    @Test
    public void testMaxRetries0() {
-      // 1 -> configure max-retries="0" and initial_server_list=localhost:port1
-      // 2 -> start 3 servers, localhost:port1, localhost:port2 and localhost:port3
-      ConfigurationBuilder builder = new ConfigurationBuilder();
-      builder.maxRetries(0);
-      RemoteCache<String, String> cache = SERVER_TEST.hotrod().withClientConfiguration(builder).withCacheMode(CacheMode.REPL_SYNC).create(0);
+      // 1 -> configure max-retries="0" and initial_server_list=server0
+      // 2 -> start 3 servers
+      RemoteCache<String, String> cache = createCache(0);
 
       // 3 -> start the client and do some operations
       for (int i = 0; i < 100; i++) {
          String random = UUID.randomUUID().toString();
          cache.put(random, random);
       }
 
-      // 4 -> kill localhost:port1
+      // 4 -> kill server0
       InetAddress killedServer = SERVERS.getServerDriver().getServerAddress(0);
       SERVERS.getServerDriver().kill(0);
 
-      // 5 -> do more operations, check that it's only connected to port2 and port3
-      // thinking one failure would be enough, but the topology is actually updated only when the client connects to one of the live servers
-      // you could have the first 2 connections to port1 and they'd both fail
-      // more than 2 is highly unlikely, but it might happen because the probability is not 0
-      for (int i = 0; i < 100; i++) {
+      // 5 -> the client will receive the topology from the server. if the client hit a server that is not working,
+      // the client topology won't be updated.
+      // we keep iterating until the client topology be: server1 and server2 or the eventually timeout occurred
+      Eventually.eventuallyEquals(2, () -> {
          String random = UUID.randomUUID().toString();
```

Comment:
`UUID.randomUUID()` is pretty heavyweight, a simple index or even `ThreadLocalRandom.current().nextInt()` would work just as well.
Also a `cache.get(k)` would be cheaper than a `cache.put(k, k)`